src/networkd.c

```diff
@@ -708,7 +710,24 @@ append_wpa_auth_conf(GString* s, const authentication_settings* auth)
     }
     if (auth->password) {
         if (auth->key_management == KEY_MANAGEMENT_WPA_PSK) {
-            g_string_append_printf(s, "  psk=\"%s\"\n", auth->password);
+            size_t len = strlen(auth->password);
+            if (len == 64) {
+                /* must be a hex-digit key representation */
+                for (unsigned i = 0; i < 64; ++i)
+                    if (!isxdigit(auth->password[i])) {
+                        g_fprintf(stderr, "ERROR: %s: PSK length of 64 is only supported for hex-digit representation\n", id);
+                        exit(1);
+                    }
+                /* this is required to be unquoted */
+                g_string_append_printf(s, "  psk=%s\n", auth->password);
+            } else if (len < 8 || len > 64) {
```

Comment:
Here we should check for `len < 8 || len > 63`, even though we already check the ` len == 64` case above, passphrase are still only valid for 8-63 characters.